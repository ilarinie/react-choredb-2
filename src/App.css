@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500');
a {
  text-decoration: none !important;
}

body {
  background: #eee;
}
.main-container {
  max-width: 900px;
  background: white;
  min-height: 100vh;
  font-family: 'Roboto', sans-serif;
  margin: 0 auto;
  border-left: 1px solid black;
  border-right: 1px solid black;
}

.table {
  width: 100%;
}

.table>thead>tr>th, .table>tbody>tr>td {
  border-bottom: 1px solid black;
  width: 50%;
  text-align: center;
}

.pagination-status {
  display: flex;
  justify-content: center;
}
.pagination-status>li{
  list-style: none;
  margin: 0 10px;
  cursor: pointer;
}
.active{
  background: black;
  color: white;
}

.App {
  text-align: center;
  font-family: 'Roboto', sans-serif;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}
.chore-link {
  color: black;
  font-weight: 500;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: black;
}

.Login {
  background: white;
  height: 100px;
  width: 90%;
  margin: 0 auto;
}


.title-container {
  padding: 20px;
}
.notification-bar {
  margin: 0 auto;
  width: 80%;
  background: lightgreen;
  border-radius: 0px 0px 10px 10px;
  text-align: center;
  color: green;
  border: 1px solid green;
  border-top: 0px solid black;
}
.create-chore-container {
  width: 90%;
  background: darkgrey;
  text-align: center;
  margin: 0 auto;
}
.new-commune-section{
  background: lightgray;
  width: 80%;
  margin: 20px auto;
  float: none;
  color: black;
  padding: 20px;
}

.username-span {
  color: #D32F2F;
}

.dashboard-container {

}

.dashboard-large-item {
  padding: 20px;
}

.loading-screen {
  text-align: center;
  width: 100%;
  height: 100vh;
  background: white;
}

.form-component-container {
  margin: 0 auto;
  width: 600px;
  padding-bottom: 100px;
  color: black;
}

.dashboard-item {
  width: 700px;
  margin: 20px 20px;
  background: white;
  padding: 20px;
}

.checkmark {
  color: green;
}

.example-enter {
  opacity: 0.01;
}

.example-enter.example-enter-active {
  opacity: 1;
  transition: opacity 50000ms ease-in;
}

.example-leave {
  opacity: 1;
}

.example-leave.example-leave-active {
  opacity: 0.01;
  transition: opacity 300ms ease-in;
}

.small-form-input {
  margin-left: 20px;
  width: 90% !important;
}
.chore-success-div, .purchase-success-div {
  color: green;
}
.chore-error-div, .purchase-error-div {
  color:#D32F2F;
}


.error-display {
  color: #D32F2F;
}
.error-display-hidden {
  display: none;
}
