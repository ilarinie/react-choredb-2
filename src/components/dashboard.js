import React, {
    Component
} from 'react';
import { Toolbar } from './toolbar';

class Dashboard extends Component {
  render() {
    return (
      <Toolbar />
    );
  }
}

module.exports = {
  Dashboard
}
