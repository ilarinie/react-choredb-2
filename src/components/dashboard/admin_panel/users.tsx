import * as React from 'react';
import {fetchUsers} from "../../../store/state_observable";
import {Card, CardHeader, CardActions, RaisedButton, CardText} from "material-ui";
import {ApiService} from "../../../services/api_service";
import {updateMessage} from "../notificator/notificator";

const UserCard = ({...props}) => {
    return (
        <Card>
            <CardHeader
                title={props.user.username}
                subtitle={props.user.user_id}
                showExpandableButton={true}
                actAsExpander={true}

            />
            <CardActions>
                { props.deleteDisabled && props.user.admin ? <span/> : <RaisedButton label="REMOVE" onClick={props.delete.bind(null, props.user)}/> }
            </CardActions>
            <CardText expandable={true}>
                <p>Has done {props.user.tasks.length} tasks.</p>
                <p>Has made {props.user.purchases.length} purchases.</p>
            </CardText>
        </Card>
    );
};

export class UserList extends React.Component<any, any> {


    constructor(props: any) {
        super(props);
        this.state = {
            users: this.props.users,
            user: this.props.user
        };
    }



    handleDelete = (user) => {
        if (confirm('Are you user you want to remove ' + user.username + '?')) {
            ApiService.send('DELETE', 'users/' + user.user_id, null, (err, res) => {
                if (!err) {
                    updateMessage("User removed succesfully.");
                    fetchUsers();
                } else {
                    updateMessage("Something went wrong");
                }
            });
        }
    }

    render() {
        if (this.state.users) {
            let users = this.state.users.map((user: any, index: any) => (<UserCard key={index} user={user} delete={this.handleDelete} deleteDisabled={user.user_id === this.state.user.user_id}/>), this);
            return (
                <div className="dashboard-large-item">
                    {users}
                </div>
            );
        } else {
            return (
                <div>Loading..</div>
            );
        }
    }


}

