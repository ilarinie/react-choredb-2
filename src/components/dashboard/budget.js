import React, {
    Component
} from 'react';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow
} from 'material-ui/Table';
import { BudgetRow } from './budget_row';



export class Budget extends Component {

    constructor(props){
      super(props);
      this.state = {
        purchases: this.props.purchases
      }
    }


    render() {
      if (this.state.purchases) {
        var budgetNodes = this.state.purchases.map( (purchase, index) => (
          <BudgetRow key={index} budgetItem={purchase} />
          ), this);
      }

      return (
        <div>
        <h2 className="dashboard-item-title">Budget</h2>
        <Table>
          <TableHeader displaySelectAll={false} adjustForCheckbox={false}>
            <TableRow>
              <TableHeaderColumn>User</TableHeaderColumn>
              <TableHeaderColumn>Spent</TableHeaderColumn>
              <TableHeaderColumn>Differential</TableHeaderColumn>
            </TableRow>
          </TableHeader>
          <TableBody>
            {this.state.purchases ? budgetNodes : <div>Loading..</div>}
          </TableBody>
        </Table>
      </div>
      )
    }
}
