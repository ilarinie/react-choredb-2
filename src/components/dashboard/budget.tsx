import * as React from 'react';
import {Table, TableBody, TableHeader, TableHeaderColumn, TableRow} from 'material-ui/Table';
import {BudgetRow} from './budget_row';

export class Budget extends React.Component < any,
any > {

  constructor(props : any) {
    super(props);
    this.state = {
      purchases: this.props.purchases
    };
  }

  render() {
    if (this.state.purchases) {
      var budgetNodes = this
        .state
        .purchases
        .map((purchase : any, index : any) => (<BudgetRow key={index} budgetItem={purchase}/>), this);
    }

    return (
      <div className="dashboard-large-item">
        <h2 className="dashboard-item-title">Budget</h2>
        <Table>
          <TableHeader displaySelectAll={false} adjustForCheckbox={false}>
            <TableRow>
              <TableHeaderColumn>User</TableHeaderColumn>
              <TableHeaderColumn>Spent</TableHeaderColumn>
              <TableHeaderColumn>Differential</TableHeaderColumn>
            </TableRow>
          </TableHeader>
          <TableBody>
            {this.state.purchases
              ? budgetNodes
              : <div>Loading..</div>}
          </TableBody>
        </Table>
      </div>
    );
  }
}
