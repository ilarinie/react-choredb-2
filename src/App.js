import React, {
    Component
} from 'react';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import './App.css';



import {Â Dashboard } from './components/dashboard/dashboard';
import { Login } from './components/login/login';

function authenticated() {
  if (localStorage.getItem('token') === null){
    return false;
  }
  return true;
}

class Base extends Component {
    render() {
      if (authenticated()){
        return (
          <div className="main-container">
            <MuiThemeProvider muiTheme={getMuiTheme(darkBaseTheme)}>
              <Dashboard />
            </MuiThemeProvider>
          </div>
        );
      } else {
        return (
          <div className="main-container">
            <MuiThemeProvider muiTheme={getMuiTheme(darkBaseTheme)}>
              <Login />
            </MuiThemeProvider>
          </div>
        );
      }
    }
}

export default Base;
