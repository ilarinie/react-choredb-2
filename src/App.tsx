import * as React from 'react';
import {MuiThemeProvider} from 'material-ui/styles';
import './App.css';
import {Â Dashboard } from './components/dashboard/dashboard';
import { Login } from './components/login/login';
import  Theme  from './theme/theme';
import {fetchCommune, getInitialState, mainStream} from "./store/state_observable";
import {State} from "./models/state";


import * as injectTapEventPlugin from 'react-tap-event-plugin';
import {LoadingScreen} from "./components/utils/loading_screen";

injectTapEventPlugin();

class App extends React.Component<{}, any> {

    sub: any;
    constructor(props: any) {
        super(props);

        this.state = {
            mainState: null
        };
    }

    componentDidMount = () => {
        this.sub = mainStream.subscribe(
            (mainState: State) => {
                this.setState({
                    mainState: mainState
                });
            }
        );
        getInitialState();
    }

    componentWillUnmount = () => {
        this.sub.dispose();
    }

    onBackButtonEvent = (e) => {
        e.preventDefault();
    }



    render() {
        let content;
        if (this.state.mainState && this.state.mainState.loggedIn) {
            content = <Dashboard mainState={this.state.mainState} />;
        } else if (!this.state.mainState) {
          content = <LoadingScreen message="Loading.."/>;
        } else {
            content = <Login />;
        }

        return (
          <div className="main-container">
            <MuiThemeProvider muiTheme={Theme}>
                {content}
            </MuiThemeProvider>
          </div>
        );
    }
}

export default App;
