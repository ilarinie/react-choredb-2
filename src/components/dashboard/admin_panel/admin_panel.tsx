import * as React from 'react';
import {UserList} from "./users";
import {NewCommuneUser} from "./new_commune_user";
import {ChoreForm} from "../chores/chore_form";
import {EditCommune} from "./edit_commune";


export class AdminPanel extends React.Component<any, any> {


    constructor(props: any) {
        super(props);
        this.state = {
            mainState: this.props.mainState
        }
    }


    render() {
        return (
            <div className="dashboard-large-item">
                <h1>Admin tools</h1>
                <h2>Add Chores</h2>
                <ChoreForm chore={{}} />
                <h2>Add Users</h2>
                <NewCommuneUser />
                <h2>Edit commune details</h2>
                <EditCommune commune={this.state.mainState.commune} />
                <h2>Remove users</h2>
                <UserList users={this.state.mainState.commune_users} user={this.state.mainState.current_user} />

            </div>
        )
    }


}