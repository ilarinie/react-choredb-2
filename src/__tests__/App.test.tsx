import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as TestUtils from 'react-dom/test-utils';
import App from '../App';
import {getMainState} from "./__mocks__/api_service";
import {Chores} from "../components/dashboard/chores/chores";
import {Budget} from "../components/dashboard/purchases/budget";
import {Dashboard} from "../components/dashboard/dashboard";
import {ChoreForm} from "../components/dashboard/chores/chore_form";
import {Login} from "../components/login/login";
import {UserList} from "../components/dashboard/admin_panel/users";
import {Profile} from "../components/dashboard/profile/profile";
import {AdminPanel} from "../components/dashboard/admin_panel/admin_panel";
import {PurchaseList} from "../components/dashboard/purchases/purchase_list";
import {NewPurchase} from "../components/dashboard/purchases/new_purchase";

;

declare var global: any;


const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
};

global.localStorage = localStorageMock;

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});


it('renders Chores', () => {
    getMainState((err, res) => {

        console.log(err);
        expect(TestUtils.createRenderer().render(
            <Chores mainState={res}/>
        )).toMatchSnapshot()
    });
})

it('renders Budget', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <Budget mainState={res}/>
        )).toMatchSnapshot()
    });
});

it('renders PurchaseList', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <PurchaseList user={res.current_user} purchases={res.purchases} />
        )).toMatchSnapshot()
    });
});

it('renders NewPurchase', () => {
        expect(TestUtils.createRenderer().render(
            <NewPurchase  />
        )).toMatchSnapshot();
});


it('renders Dashboard', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <Dashboard mainState={res}/>
        )).toMatchSnapshot()
    });
});

it('renders ChoreForm', () => {
    expect(TestUtils.createRenderer().render(
        <ChoreForm chore={{}}/>
    )).toMatchSnapshot()
});
it('renders ChoreForm with chore', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <ChoreForm chore={res.chores[0]}/>
        )).toMatchSnapshot()
    });
});

it('renders Login', () => {
    expect(TestUtils.createRenderer().render(
        <Login />
    )).toMatchSnapshot()
});

it('renders UserList', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <UserList mainState={res} />
        )).toMatchSnapshot()
    });
})

it('renders Menu', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <UserList mainState={res} />
        )).toMatchSnapshot()
    });
})

it('renders Profile', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <Profile mainState={res} />
        )).toMatchSnapshot()
    });
})

it('renders Admin Panel', () => {
    getMainState((err, res) => {
        expect(TestUtils.createRenderer().render(
            <AdminPanel mainState={res} />
        )).toMatchSnapshot()
    });
})