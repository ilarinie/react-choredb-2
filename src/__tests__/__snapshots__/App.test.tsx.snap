// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ChoreForm 1`] = `
<div
  className="form-component-container"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Create a new chore
    </h1>
    <Paper
      circle={false}
      rounded={true}
      transitionEnabled={true}
      zDepth={2}
    >
      <TextField
        className="small-form-input"
        defaultValue={undefined}
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Chore name"
        id="chore-form-name"
        multiLine={false}
        rows={1}
        type="text"
        underlineShow={false}
      />
      <Divider
        inset={false}
      />
      <TextField
        className="small-form-input"
        defaultValue={undefined}
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="How often this should be done? (hours)"
        id="chore-form-priority"
        multiLine={false}
        rows={1}
        type="number"
        underlineShow={false}
      />
      <Divider
        inset={false}
      />
      <TextField
        className="small-form-input"
        defaultValue={undefined}
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Points awarded"
        id="chore-form-points"
        multiLine={false}
        rows={1}
        type="number"
        underlineShow={false}
      />
      <Divider
        inset={false}
      />
    </Paper>
    <br />
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="Create"
      labelPosition="after"
      primary={false}
      secondary={false}
      type="submit"
    />
    <span />
    <div
      className="chore-error-div"
    >
      <p />
    </div>
    <div
      className="chore-success-div"
    >
      <p />
    </div>
  </form>
</div>
`;

exports[`renders Login 1`] = `
<div
  className="login-container"
>
  <div
    className="banner"
  >
    <h1>
      ChoreDB 2.0
    </h1>
  </div>
  <Tabs
    initialSelectedIndex={0}
    onChange={[Function]}
    value="login"
  >
    <Tab
      label="Log in"
      value="login"
    >
      <div
        className="login-container"
      >
        <div>
          <h1>
            Please log in
          </h1>
          <form
            onSubmit={[Function]}
          >
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={false}
              hintText="Username"
              id="username"
              multiLine={false}
              rows={1}
              type="text"
              underlineShow={true}
            />
            <br />
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={false}
              hintText="Password"
              id="password"
              multiLine={false}
              rows={1}
              type="password"
              underlineShow={true}
            />
            <br />
            <div
              className="auth-failure"
            />
            <RaisedButton
              disabled={false}
              fullWidth={true}
              label="Log in"
              labelPosition="after"
              primary={false}
              secondary={false}
              type="submit"
            />
          </form>
        </div>
      </div>
    </Tab>
    <Tab
      label="Register"
      value="register"
    >
      <Register />
    </Tab>
  </Tabs>
</div>
`;

exports[`renders NewPurchase 1`] = `
<div
  className="form-component-container"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Create a new purchase
    </h1>
    <Paper
      circle={false}
      rounded={true}
      transitionEnabled={true}
      zDepth={2}
    >
      <TextField
        className="small-form-input"
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Description"
        id="purchase-description-input"
        multiLine={false}
        rows={1}
        type="text"
        underlineShow={false}
      />
      <Divider
        inset={false}
      />
      <TextField
        className="small-form-input"
        disabled={false}
        errorText=""
        floatingLabelFixed={false}
        fullWidth={false}
        hintText="Amount"
        id="purchase-amount-input"
        multiLine={false}
        rows={1}
        type="tel"
        underlineShow={false}
      />
      <Divider
        inset={false}
      />
    </Paper>
    <br />
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="Create"
      labelPosition="after"
      primary={false}
      secondary={false}
      type="submit"
    />
    <div
      className="purchase-success-div"
    />
  </form>
</div>
`;
